### PR TITLE
Examples fixes and preview.astro.new support

### PR DESCRIPTION
**Pull request by delucis**
## Changes

- This PR (partially) fixes support for `base` in the blog, portfolio, and with-nanostores examples.
  - In the blog and portfolio examples, this fixes the highlighting of the current page in the site navigation components. It doesn’t fix other aspects of these templates to work with base (e.g. image paths etc.) just for simplicity’s sake, but we could in a follow up if we wanted.
  - In the with-nanostores example, this fixes all user-authored links to use a base.
- This improves the deployments to preview.astro.new, which builds each example with `base` and then tries to auto-prefix with base when it can, but there are some edge cases it can’t handle, which this PR fixes
- This PR also adds a new GitHub workflow which runs when pushing to `main`. It checks to see if anything in the `examples/` directory has changed, and if so sends a request to Netlify to redeploy preview.astro.new
- This workflow requires a new `NETLIFY_PREVIEWS_BUILD_HOOK` secret adding to this repo. @ematipico added this ✅ 

## Testing

- Ran the examples locally with and without `base` set

## Docs

n/a bug fix only

### CODE REVIEW

File: .github/workflows/examples-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Check if examples changed
+        id: changes
+        uses: dorny/paths-filter@v3
+        with:
+          filters: |
+            examples:
+              - 'examples/**'
```
Comment (ematipico) [inline .github/workflows/examples-deploy.yml:24]: You don't need an action for that, Github already supports that:

https://github.com/withastro/astro/blob/65e50eb7b6d7b10a193bba7d292804ac0e55be18/.github/workflows/check.yml#L10-L16

File: .github/workflows/examples-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Notify Netlify
+        if: ${{ steps.changes.outputs.examples == 'true' }}
+        run: 'curl -X POST -d {} ${{ env.BUILD_HOOK }}'
```
Comment (ematipico) [inline .github/workflows/examples-deploy.yml:27]: You might want add a comment explaining what this does

**Review by ematipico — APPROVED** [review]
I didn't checkout the branch to check the UI changes. I left a couple of suggestions for the workflow.

File: .github/workflows/examples-deploy.yml (hunk 2 above)
Comment (delucis) [reply .github/workflows/examples-deploy.yml:20]: Added a bit more detail in the `name` which hopefully should cover it along with the summary at the top of the file.

**Pull request merged**